Sources/Screens/RepositoryList/RepositoryListViewModel.swift

```diff
@@ -25,12 +24,17 @@ extension RepositoryListView {
         func fetchData() {
             Task {
                 do {
-                    repositories = try await apiManager
-                        .getRepositories(for: searchTerm)
+                    repositories = try await ApiManager.shared.fetch(
+                        type: Root.self,
+                        endpoint: ApiManager.Endpoint
```

Comment:
instead of explicitly typing _ApiManager.Endpoint.searchRepositories_ you should be able to just type _.searchRepositories_